student_explorer/backends.py

```diff
@@ -1,18 +1,38 @@
 from djangosaml2.backends import Saml2Backend
 from django.core.exceptions import PermissionDenied
+from django.conf import settings
+
 import logging
 
 logger = logging.getLogger(__name__)
 
 
 class ActiveUserOnlySAML2Backend(Saml2Backend):
     def authenticate(self, **kwargs):
-        user = super(ActiveUserOnlySAML2Backend, self).authenticate(**kwargs)
+        user = None
+        try:
+            user = super(ActiveUserOnlySAML2Backend, self).authenticate(**kwargs)
+        except PermissionDenied:
+            # If user 
+            if user:
+                user.is_active = False
+                user.save()
+            raise PermissionDenied
+        # If the user doesn't exist for some reason return permission denied
         if not user:
             raise PermissionDenied
-        if user.is_active:
-           return user
+        # The user should be made active if they exist and aren't active
+        if not user.is_active:
+           user.is_active = True
+           user.save()
+        return user
+            
+
+    def is_authorized(self, attributes, attribute_mapping):
+        # If there are any groups that we're a member of, return true
+        if attributes.get('isMemberOf'):
```

Comment:
Is the SAML only return the member status for request group? That why we can check for "any group member" status?

Reply:
I asked for clarification about this on the ServiceNow ticket but the way I'm seeing it is if I'm in this specific group I requested 
`attributes.get('isMemberOf') == ['Student Explorer Users']
`
When I was removed from the group
`attributes.get('isMemberOf') == [] `

So we don't have to check anything locally